doc/design/send_recv.md

```diff
@@ -0,0 +1,68 @@
+# Send and Recv OP
+
+With PaddlePaddle Fluid, the node to node communication is done
+through the send and recv OP. The send and recv OP will be
+automatically added to
+the
+[`ProgramDesc`](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/program.md#blockdesc-and-programdesc) by
+the
+[distributed transpiler](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/dist_refactor/distributed_architecture.md#distributed-transpiler) when
+a local training `ProgramDesc` is transpiled to the distributed
+training `ProgramDesc`.
+
+
+## Send OP
+
+The send OP sends its input tensors to one or multiple recv OPs and
+then fetches its output tensors from the recv OPs.
+
+|Input|Description|
+|------|------|
+|X|Input tensor to be sent|
+
+|Output|Description|
+|------|------|
+|Out|Output tensor to be received from the recv OP|
+
+|Attribute|Description|
+|------|------|
+|endpoints|Endpoints to send the variables to|
+|epmap|Endpoints in the order of input variables, used for sharding|
+
+
+## Recv OP
+
+The recv OP receives its input tensors, runs the optimization block, and
+serves the tensors requested by the send OP.
+
+|Input|Description|
+|------|------|
+|RX|Input tensor to be received|
+
+|Attribute|Description|
```

Comment:
Where in this setup does the receive OP get the list of tensors as an input from send OP ?
Or would we have one receive OP per parameter ?

Reply:
The input of send of will be the same as the input of recv op, if there is only 1 recv op. They are all setup by the transpiler.

Currently there is one send op per trainer, and one recv op per pserver. We will move to one send/recv pair per parameter in the future.

Reply:
Thanks so much, that helps!